### PR TITLE
wasmedge: add version 0.11.1

### PR DESCRIPTION
**Pull request by KingKili**
Specify library name and version:  **wasmedge/0.11.1**
---

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Review by toge — APPROVED** [review]

File: recipes/wasmedge/all/conandata.yml [hunk 1]
```diff
@@ -1,4 +1,25 @@
 sources:
+  "0.11.1":
+    Windows:
+      "x86_64":
+        Visual Studio:
+          - url: "https://github.com/WasmEdge/WasmEdge/releases/download/0.11.1/WasmEdge-0.11.1-windows.zip"
```
Comment (prince-chrismc) [inline recipes/wasmedge/all/conandata.yml:6]: Seems a little overkill this is tied to the compiler 🤔 

usually binaries we just delete the compiler so this is odd

File: recipes/wasmedge/all/conanfile.py [hunk 2]
```diff
@@ -41,29 +43,37 @@ def package_id(self):
...
 
         srclibdir = os.path.join(self._source_subfolder, "lib64" if self.settings.os == "Linux" else "lib")
         srcbindir = os.path.join(self._source_subfolder, "bin")
-
-        self.copy("wasmedge_c.lib", src=srclibdir, dst="lib", keep_path=False)
-        self.copy("wasmedge_c.dll", src=srcbindir, dst="bin", keep_path=False)
-        self.copy("libwasmedge_c.so*", src=srclibdir, dst="lib", keep_path=False)
-        self.copy("libwasmedge_c.dylib", src=srclibdir,  dst="lib", keep_path=False)
+        if Version(self.version) >= "0.11.1":
+            self.copy("wasmedge.lib", src=srclibdir, dst="lib", keep_path=False)
```
Comment (prince-chrismc) [inline recipes/wasmedge/all/conanfile.py:58]: Would be nice to sneak in `conan.tools.files.copy` here to get some more progress updating it 🙏

**Review by prince-chrismc — APPROVED** [review]
I think this recipe needs a good cleanup but I have no issue with the changes being made

I'd love to see a follow up PR improving it

**Comment by robberphex** [comment]
I just noticed that wasmedge cannot get on macOS?

**Comment by prince-chrismc** [comment]
> I just noticed that wasmedge cannot get on macOS?

If you know who/where please open a PR it has not been supported since https://github.com/KingKili/conan-center-index/blob/d3202dde84c4eaa9c3bb2f263f4fc7ce166bc352/recipes/wasmedge/all/conandata.yml#L86

The community + maintainers (like me) dont always know the project in depth to notice it's missing -- let alone how to fix it

So your help would be amazing!

**Pull request merged**